digdag-tests/src/test/java/acceptance/RequireIT.java

```diff
@@ -185,6 +203,159 @@ public void testIgnoreProjectIdParam()
+                "-c", config.toString(),
+                "-e", server.endpoint());
+        assertThat(pushChildStatus.errUtf8(), pushChildStatus.code(), is(0));
+
+        // extract child project id
+        Matcher m = Pattern.compile(".*\\s+id:\\s+(\\d+).*").matcher(pushChildStatus.outUtf8());
+        assertThat(m.find(), is(true));
+        String childProjectId = m.group(1);
+
+        // Push parent project
+        Files.write(projectDir.resolve("parent_another_project.dig"), asList(Resources.toString(
+                Resources.getResource("acceptance/require/parent_another_project.dig"), UTF_8)
+                .replace("__CHILD_PROJECT_ID__", childProjectId)
+                .replace("__CHILD_PROJECT_NAME__", childProjectName)));
+        CommandStatus pushParentStatus = main("push",
+                "--project", projectDir.toString(),
+                "parent_another",
+                "-c", config.toString(),
+                "-e", server.endpoint());
+        assertThat(pushParentStatus.errUtf8(), pushParentStatus.code(), is(0));
+
+        CommandStatus startStatus = main("start",
+                "-c", config.toString(),
+                "-e", server.endpoint(),
+                "parent_another", "parent_another_project",
+                "--session", "now"
+        );
+        assertThat(startStatus.errUtf8(), startStatus.code(), is(0));
+    }
+
+    @Test
+    public void testRerunOnParam()
+            throws Exception
+    {
+        // Create new project
+        CommandStatus initStatus = main("init",
+                "-c", config.toString(),
+                projectDir.toString());
+        assertThat(initStatus.errUtf8(), initStatus.code(), is(0));
+
+        Path childOutFile = projectDir.resolve("child.out").toAbsolutePath().normalize();
+        prepareForChildWF(childOutFile);
+        copyResource("acceptance/require/parent_rerun_on.dig", projectDir.resolve("parent.dig"));
+        copyResource("acceptance/require/child_rerun_on.dig", projectDir.resolve("child.dig"));
+
+        // Push the project
+        CommandStatus pushStatus = main("push",
+                "--project", projectDir.toString(),
+                "require",
+                "-c", config.toString(),
+                "-e", server.endpoint(),
+                "-r", "4711");
+        assertThat(pushStatus.errUtf8(), pushStatus.code(), is(0));
+
+        // test for rerun_on: none
+        {
+            String sessionTime = "2020-05-28 12:34:01";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "none", false);
+            assertThat("Number of child's attempt must be one. (== require> skip the call)", childAttempts.getAttempts().size(), is(1));
+        }
+
+        // test for rerun_on: all. require> kick child always.
+        {
+            String sessionTime = "2020-05-28 12:34:11";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "all", false);
```

Comment:
How about adding a test case of the combination of `all` and true?

Reply:
I added it. Thanks!